### PR TITLE
feat(optimus): change how secrets are used by plugins

### PR DESCRIPTION
**Pull request by sbchaos**

### CODE REVIEW

File: odpf/optimus/plugins/v1beta1/dependency_resolver.proto [hunk 1]
```diff
@@ -21,10 +21,17 @@ service DependencyResolverModService {
     rpc GenerateDependencies(GenerateDependenciesRequest) returns (GenerateDependenciesResponse);
 }
 
+message PluginJobConfigs {
```
Comment (sravankorumilli) [inline odpf/optimus/plugins/v1beta1/dependency_resolver.proto:24]: why do we need a segregation between secrets and regular configs, plugin implementation knows the difference

File: odpf/optimus/plugins/v1beta1/dependency_resolver.proto [hunk 2]
```diff
@@ -21,10 +21,17 @@ service DependencyResolverModService {
...
+
 message GenerateDestinationRequest {
-    Configs config = 1;
-    Assets assets = 2;
-    odpf.optimus.core.v1beta1.ProjectSpecification project = 3;
+    Configs config = 1  [deprecated=true];
+    Assets assets = 2  [deprecated=true];
+    odpf.optimus.core.v1beta1.ProjectSpecification project = 3  [deprecated=true];
+
+    PluginJobConfigs job_configs = 4;
```
Comment (sravankorumilli) [inline odpf/optimus/plugins/v1beta1/dependency_resolver.proto:34]: JobConfig & Assets we need to pass and we are passing those details, we don't need to deprecate assets.
PluginJobConfigs is not necessary imo we can leverage configs itself

File: odpf/optimus/plugins/v1beta1/dependency_resolver.proto [hunk 3]
```diff
@@ -34,9 +41,11 @@ message GenerateDestinationResponse {
 }
 
 message GenerateDependenciesRequest {
-    Configs config = 1;
+    Configs config = 1  [deprecated=true];
     Assets assets = 2;
-    odpf.optimus.core.v1beta1.ProjectSpecification project = 3;
+    odpf.optimus.core.v1beta1.ProjectSpecification project = 3  [deprecated=true];
+
+    PluginJobConfigs job_configs = 4;
```
Comment (sravankorumilli) [inline odpf/optimus/plugins/v1beta1/dependency_resolver.proto:48]: similar to the above comment

File: odpf/optimus/plugins/v1beta1/dependency_resolver.proto (hunk 1 above)
Comment (sbchaos) [reply odpf/optimus/plugins/v1beta1/dependency_resolver.proto:24]: Sure, will remove this

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**